### PR TITLE
Add a new Equivariant GNN named `ViSNet`

### PR DESCRIPTION
**Pull request by XJTUNR**
Hi rusty1s,

We have implemented the ViSNet model as described in the [paper](https://arxiv.org/abs/2210.16518) and would like to propose a pull request to merge our implementation into the PyG. The complete repository, including the training code, can be found at this [link](https://github.com/microsoft/AI2BMD/tree/ViSNet).

We have previously contacted you by email to discuss this PR, and we believe that this implementation would be a valuable addition to the PyG library. 

Please let us know if you have any questions or suggestions for improvements. We would be happy to make any necessary changes to ensure the quality and consistency of the codebase.

Thank you for considering our pull request.

Best regards,
Yusong

### CODE REVIEW

File: torch_geometric/nn/models/__init__.py [hunk 1]
```diff
@@ -25,45 +25,15 @@
...
-    'PMLP',
-    'NeuralFingerprint',
+    'MLP', 'GCN', 'GraphSAGE', 'GIN', 'GAT', 'PNA', 'EdgeCNN',
+    'JumpingKnowledge', 'MetaLayer', 'Node2Vec', 'DeepGraphInfomax',
+    'InnerProductDecoder', 'GAE', 'VGAE', 'ARGA', 'ARGVA', 'SignedGCN',
+    'RENet', 'GraphUNet', 'SchNet', 'DimeNet', 'DimeNetPlusPlus',
+    'to_captum_model', 'to_captum_input', 'captum_output_to_dicts',
+    'MetaPath2Vec', 'DeepGCNLayer', 'TGNMemory', 'LabelPropagation',
+    'CorrectAndSmooth', 'AttentiveFP', 'RECT_L', 'LINKX', 'LightGCN',
+    'MaskLabel', 'GroupAddRev', 'GNNFF', 'PMLP', 'NeuralFingerprint', 'ViSNet'
```
Comment (akihironitta) [inline torch_geometric/nn/models/__init__.py:38]:
```suggestion
    'MaskLabel', 'GroupAddRev', 'GNNFF', 'PMLP', 'NeuralFingerprint', 'ViSNet',
```

**Review by akihironitta — COMMENTED** [review]
Would you mind adding docstrings just like other models have in PyG, e.g., https://github.com/pyg-team/pytorch_geometric/blob/2.4.0/torch_geometric/nn/models/graph_mixer.py? It'd help PyG users and reviewers a lot :)

**Comment by XJTUNR** [comment]
hi [akihironitta](https://github.com/akihironitta), I've included a docstring in the code, but it seems to be causing the document compilation to fail. I'm not well-versed with docstrings, could you assist me in modifying it?

**Comment by XJTUNR** [comment]
> Would you mind adding docstrings just like other models have in PyG, e.g., https://github.com/pyg-team/pytorch_geometric/blob/2.4.0/torch_geometric/nn/models/graph_mixer.py? It'd help PyG users and reviewers a lot :)

Hi [akihironitta](https://github.com/akihironitta), I've included a docstring in our code and successfully built documentation.

File: torch_geometric/nn/models/visnet.py [hunk 2]
```diff
@@ -0,0 +1,1345 @@
...
+        max_z (int): The maximum atomic numbers.
+        cutoff (float): The cutoff distance.
+        max_num_neighbors (int):The maximum number of neighbors
+            considered for each atom.
+        vertex (bool): Whether to use vertex geometric features.
+    """
+    def __init__(
+        self,
+        lmax: int = 1,
+        vecnorm_type: str = "none",
```
Comment (EdisonLeeeee) [inline torch_geometric/nn/models/visnet.py:931]: `None` or `"none"` need to be consistent.

File: torch_geometric/nn/models/visnet.py [hunk 3]
```diff
@@ -0,0 +1,1345 @@
...
+        num_heads: int = 8,
+        num_layers: int = 9,
+        hidden_channels: int = 256,
+        num_rbf: int = 32,
+        trainable_rbf: bool = False,
+        max_z: int = 100,
+        cutoff: float = 5.0,
+        max_num_neighbors: int = 32,
+        vertex: bool = False,
+    ) -> None:
```
Comment (EdisonLeeeee) [inline torch_geometric/nn/models/visnet.py:942]:
```suggestion
    ):
```

File: torch_geometric/nn/models/visnet.py [hunk 4]
```diff
@@ -0,0 +1,1345 @@
...
+        self.embedding.reset_parameters()
+        self.distance_expansion.reset_parameters()
+        self.neighbor_embedding.reset_parameters()
+        self.edge_embedding.reset_parameters()
+        for layer in self.vis_mp_layers:
+            layer.reset_parameters()
+        self.out_norm.reset_parameters()
+        self.vec_out_norm.reset_parameters()
+
+    def forward(self, z, pos, batch):
```
Comment (EdisonLeeeee) [inline torch_geometric/nn/models/visnet.py:1000]: type hints

**Review by rusty1s — APPROVED** [review]

**Comment by rusty1s** [comment]
Thanks for the review @EdisonLeeeee. Incorporated :)

**Pull request merged**